pkg/services/ngalert/schedule/schedule.go

```diff
@@ -431,18 +453,24 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 	for {
 		select {
 		// used by external services (API) to notify that rule is updated.
-		case lastVersion := <-updateCh:
+		case ctx := <-updateCh:
+			// we can continue directly if the rule is paused because in the next tick it will be picked up. The state
+			// will be cleaned up and the routine will be stopped and removed from the registry
+			if ctx.IsPaused {
+				logger.Info("Skip updating rule because it is paused")
+				continue
+			}
```

Comment:
I do not think this is correct.  This creates a difference in behavior of the update operation:
1. rule is not paused. The state of the rule is cleared synchronously to API request. In the UI user sees the state cleaned up immediately.
2. rule is paused. The state of the rule will be reset in up to 10 seconds.

I do not think this is a good behavior. Instead, I think it should call clearState with paused reason.